### ISSUE TITLE
Feature: Add save and load functions in cache 

### ISSUE THREAD

**Issue by alidevhere** [Code-Hex/go-generics-cache#41]
Hii !!
It would be greate to have data persistance. I would like to contribute to add save and load feature.
I am thinking of using gob to save / retreive data to / from file.

**Comment by Code-Hex** [comment]
@alidevhere Thanks
What do you think for the usage?

**Comment by alidevhere** [comment]
@Code-Hex I need to add graceful shutdown for my server. If my server restarts it should save whatever in cache and reload when started again. 
So if a make little fix and restart it doesnot affect other processes going on.

**Comment by Code-Hex** [comment]
@alidevhere I see. sounds good.

I think that the development direction should be as follows:

1. Create new [options](https://github.com/Code-Hex/go-generics-cache/blob/9f98378e35c0f1d37a62f9e0b34f9234a0007e1c/cache.go#L113) like `WithSave`, `WithRestore`.
2. `WithSave` may have a functional parameter like `func(cache Interface) {}` and this function will be called received context.Done.
3. `WithRestore` may have a functional parameter like `func(cache Interface) {}` and this function will be initialized.

Is it possible to contribute in this way?

**Comment by alidevhere** [comment]
@Code-Hex  Yeah sure!!

I have created a proof of concept pull [request](https://github.com/Code-Hex/go-generics-cache/pull/42), I tried to Add Save and Load functions for cache. Let me share my findings with you:

1. I used gob and gob needs Struct fields to be exported/public to encode and decode them.
2. It is 100% working with simple cache. But for other algorithm caches we use datastructures like linked list etc. In these cases it is not as straight forward as for simple cache. We will have to do some manual work.

@Code-Hex  I will do some more work on it with directions you gave. 

Thanks

**Comment by Code-Hex** [comment]
@alidevhere Thanks your initial PoC but I'm sorry I cannot accept the PR 🙏

**Comment by alidevhere** [comment]
> @alidevhere Thanks your initial PoC but I'm sorry I cannot accept the PR 🙏

Yes, I made this PR just to show POC i will make a complete PR later
